lib/tgspam/detector.go

```diff
@@ -745,3 +758,10 @@ func (d *Detector) cleanText(text string) string {
 	}
 	return result.String()
 }
+
+func (d *Detector) ctxWithStoreTimeout() (context.Context, context.CancelFunc) {
```

Comment:
We can add CheckWithCtx which would be called from Check with existing signature and that param, and otherwise use provided context.